crates/turbopack-dev-server/src/source/conditional.rs

```diff
@@ -84,15 +84,19 @@ impl ContentSource for ConditionalContentSource {
         data: turbo_tasks::Value<ContentSourceData>,
     ) -> Result<ContentSourceResultVc> {
         let first = self.activator.get(path, data.clone());
-        if let ContentSourceResult::NotFound = &*first.await? {
-            if self.is_activated() {
-                Ok(self.action.get(path, data))
-            } else {
-                Ok(first)
+        if let ContentSourceContent::NotFound = &*first.await?.content.await? {
+            if !self.is_activated() {
+                return Ok(first);
             }
-        } else {
-            self.activate();
+        }
+        self.activate();
+        let second = self.action.get(path, data);
+        let first_specificity = first.await?.specificity.await?;
+        let second_specificity = second.await?.specificity.await?;
+        if first_specificity >= second_specificity {
```

Comment:
Right now we consider a higher specificity to be greater than another specificity. I think doing the inverse might make more sense, e.g. when sorting an array of specificities, I think the default order should be from highest to lowest, so the iteration order makes sense.